sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -253,7 +252,7 @@ case class FilterEstimation(plan: Filter) extends Logging {
       1.0 - nullPercent
     }
 
-    Some(percent)
+    Some(percent.toDouble)
```

Comment:
The calculation of `percent` should use double too

Reply:
This is because `nullPercent` is calculated by `colStat.nullCount / rowCount`, both `nullCount` and rowCount` are BigInt, and need to be converted to BigDecimal..